cmd/homi/setup/cmd.go

```diff
@@ -1065,22 +1092,25 @@ func WriteFile(content []byte, parentFolder string, fileName string) {
 }
 
 func findGenType(ctx *cli.Context) int {
+	// Klaytn-Node: genTypeFlag's default value is docker
+	genTypeFlag := ctx.String(genTypeFlag.Name)
 	genType := TypeNotDefined
-	if len(ctx.Args()) >= 1 {
+	if ctx.Args().Present() {
+		genTypeFlag = ctx.Args()[0]
 		for i, t := range Types {
-			if t == ctx.Args()[0] {
+			if t == genTypeFlag {
 				genType = i
 				break
 			}
 		}
-		if genType == TypeNotDefined {
-			fmt.Printf("Wrong Type : %s\nSupported Types : [docker, local, remote, deploy]\n\n", ctx.Args()[0])
-			cli.ShowSubcommandHelp(ctx)
-			os.Exit(1)
-		}
 	} else {
 		genType = TypeDocker
 	}
+	if genType == TypeNotDefined {
+		fmt.Printf("Wrong Type : %s\nSupported Types : [docker, local, remote, deploy]\n\n", genTypeFlag)
+		cli.ShowSubcommandHelp(ctx)
+		os.Exit(1)
+	}
```

Comment:
```suggestion
func indexGenType(genTypeFlag string) int {
	for typeIndex, typeString := range Types {
		if genTypeFlag == typeString {
			return typeIndex
		}
	}
	return TypeNotDefined
}

func findGenType(ctx *cli.Context) int {
	var genType int
	if ctx.Args().Present() {
		genType = indexGenType(ctx.Args()[0])
	} else {
		genType = indexGenType(ctx.String(genTypeFlag.Name))
	}
	
	if genType == TypeNotDefined {
		fmt.Printf("Wrong Type : %s\nSupported Types : [docker, local, remote, deploy]\n\n", genTypeFlag)
		cli.ShowSubcommandHelp(ctx)
		os.Exit(1)
	}
	return genType
}
``` 
I think this case skips the line "genTypeFlag := ctx.String(genTypeFlag.Name)" and does not check genTypeFlag.  
So I suggest considering your initial intention that the first argument should have a higher precedence than genTypeFlag

Reply:
Clean code :) Thanks!